packages/smart-contracts/scripts-create2/contract-setup/setupErc20ConversionProxy.ts

```diff
@@ -0,0 +1,43 @@
+import { erc20ConversionProxy } from '../../src/lib';
+import { HardhatRuntimeEnvironmentExtended } from '../types';
+import utils from '@requestnetwork/utils';
+import { updateChainlinkConversionPath, updatePaymentErc20FeeProxy } from './adminTasks';
+
+/**
+ * Updates the values of the batch fees of the BatchPayments contract, if needed
+ * @param contractAddress address of the BatchPayments Proxy
+ * @param hre Hardhat runtime environment
+ */
+export const setupErc20ConversionProxy = async (
+  contractAddress: string,
+  hre: HardhatRuntimeEnvironmentExtended,
+): Promise<void> => {
+  // Setup contract parameters
+  const Erc20ConversionProxyContract = new hre.ethers.Contract(
+    contractAddress,
+    erc20ConversionProxy.getContractAbi(),
+  );
+  await Promise.all(
+    hre.config.xdeploy.networks.map(async (network) => {
+      try {
+        let provider;
+        if (network === 'celo') {
+          provider = utils.getCeloProvider();
+        } else {
+          provider = utils.getDefaultProvider(network);
+        }
+        const wallet = new hre.ethers.Wallet(hre.config.xdeploy.signer, provider);
+        const signer = wallet.connect(provider);
```

Comment:
isn't this equivalent?
```suggestion
        const signer = new hre.ethers.Wallet(hre.config.xdeploy.signer).connect(provider);
```

Reply:
Right.
I've made some factorization.